orangecontrib/text/widgets/owpubmed.py

```diff
@@ -0,0 +1,463 @@
+            self.pubmed_api = Pubmed(
+                    email=email,
+                    progress_callback=self.api_progress_callback,
+                    error_callback=self.api_error_callback,
+            )
+
+        if self.search_tabs.currentIndex() == 0:
+            # Get query parameters.
+            terms = self.keyword_combo.currentText().split()
+            authors = self.author_input.text().split()
+
+            # If no keywords, alert that the query is too vague.
+            if not terms:
+                self.warning(0, 'Please specify the keywords for this query.')
+                self.run_search_button.setEnabled(True)
+                self.retrieve_records_button.setEnabled(True)
+                return
+
+            # Check date formatting.
+            pdate_from_flag = self.pub_date_from and not eval_date(
+                    self.pub_date_from
+            )
+            pdate_to_flag = self.pub_date_to and not eval_date(
+                    self.pub_date_to
+            )
+            if pdate_from_flag or pdate_to_flag:
+                self.warning(0, 'Please specify dates with digits in '
+                                'YYYY/MM/DD format.')
+                self.run_search_button.setEnabled(True)
+                self.retrieve_records_button.setEnabled(True)
+                return
+
+            error = self.pubmed_api._search_for_records(terms,
+                                                        authors,
+                                                        self.pub_date_from,
+                                                        self.pub_date_to)
+            if error is not None:
+                self.warning(0, str(error))
+                return
+
+            if self.keyword_combo.currentText() not in self.recent_keywords:
+                self.recent_keywords.insert(
+                        0,
+                        self.keyword_combo.currentText()
+                )
+        else:
+            query = self.advanced_query_input.toPlainText()
+            if not query:
+                self.warning(0, 'Please specify the keywords for this query.')
+                self.run_search_button.setEnabled(True)
+                self.retrieve_records_button.setEnabled(True)
+                return
+            error = self.pubmed_api._search_for_records_advanced(query)
+
+            if error is not None:
+                self.warning(0, str(error))
+                return
+
+            self.last_advanced_query = query
+
+        self.enable_controls()
+        self.update_search_info()
+
+    def retrieve_records(self):
+        """
+        Retrieves the records that were queried with '_search_for_records()'.
+        If retrieval was successful, generates a corpus with the text fields as
+        meta attributes.
+        :param num_records: The number of records to retrieve.
+        :type num_records: int
+        :return: orangecontrib.text.corpus.Corpus
```

Comment:
Please write [Napoleon-style docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/index.html) in new code.